### PR TITLE
Add an optional click track to mm.Recorder

### PR DESCRIPTION
**Pull request by notwaldorf**
Fixes https://github.com/tensorflow/magenta-js/issues/130

### CODE REVIEW

**Comment by notwaldorf** [comment]
Hey @tambien does this look ok re: how I am using `Tone.Loop`? Not super familiar with `Tone.js` so I want to make sure I'm not being a dummy and there's a better way

File: music/src/core/recorder.ts [hunk 1]
```diff
@@ -82,6 +122,55 @@ export class Recorder {
...
+   * @param countIn True if it should play a count in.
+   */
+  enablePlayCountIn(countIn: boolean) {
+    this.config.playCountIn = countIn;
+  }
+
+  private initClickLoop() {
+    let clickStep = 0;
+    this.clickLoop = new Tone.Loop((_:number) => {
+      if (clickStep % 4 === 0) {
```
Comment (adarob) [inline music/src/core/recorder.ts]: Add TODO about time sig.

File: music/src/core/recorder.ts [hunk 2]
```diff
@@ -82,6 +122,57 @@ export class Recorder {
...
+    this.config.playCountIn = countIn;
+  }
+
+  private initClickLoop() {
+    let clickStep = 0;
+    this.clickLoop = new Tone.Loop((_:number) => {
+      // TODO(notwaldorf): It would be nice if this took into account a
+      // time signature.
+      if (clickStep % 4 === 0) {
+        this.loClick.triggerAttack('G5');
```
Comment (tambien) [inline music/src/core/recorder.ts]: Using the passed in number parameter (_:number)  as the second argument to triggerAttack will give you much more accurate timing.

Comment (notwaldorf) [reply music/src/core/recorder.ts]: I'm not sure I understand how to get "every 4 beats" from that, though

**Pull request merged**